server/src/main/java/org/elasticsearch/transport/RemoteClusterPortSettings.java

```diff
@@ -130,6 +131,14 @@ public class RemoteClusterPortSettings {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<ByteSizeValue> MAX_HEADER_SIZE = Setting.byteSizeSetting(
+        REMOTE_CLUSTER_PREFIX + "max_header_size",
```

Comment:
We may want to call this `max_request_header_size` because it applies to only incoming requests but _not_ responses, i.e. the remote cluster server channel does _not_ handle incoming responses.

Reply:
Pushed [b86cfa0](https://github.com/elastic/elasticsearch/pull/98692/commits/b86cfa0e27e16d2cdeccf191a8c4b79601df82d8)